### ISSUE TITLE
ganeti should zero metadata device before calling drbdmeta

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#238]
Originally reported of Google Code with ID 182.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

gnt-cluster --version
gnt-cluster (ganeti v2.4.2) 2.4.2

gnt-cluster version
Software version: 2.4.2
Internode protocol: 2040000
Configuration format: 2040000
OS api version: 20
Export interface: 0


<b>What distribution are you using?</b>
gentoo amd64

<b>What steps will reproduce the problem?</b>

the problem is not easyly reproduceable. i've run into this issue on a ganeti cluster on which many instances where created and removed within a short time. while changing the disk template of an existing instance from plain to drbd it got the following error:

Mon Jul 18 17:06:42 2011 Initializing DRBD devices...
 Failure: command execution error:
 Can't create block device
 <DRBD8(hosts=node1.domain.intern/6-node3.domain.intern/2, port=11088,
 configured as 192.168.182.82:11088 192.168.182.80:11088,
 backend=<LogicalVolume(/dev/system/1dc9c4e3-7cd3-4bf4-a9e9-882392c5a2ec.disk0_data,
 not visible, size=10240m)>,
 metadev=<LogicalVolume(/dev/system/1dc9c4e3-7cd3-4bf4-a9e9-882392c5a2ec.disk0_meta,
 not visible, size=128m)>, visible as /dev/disk/0, size=10240m)> on node
 node3.frigosped.intern for instance intra01: Can't create block device:
 Can't initialize meta device: md_offset 0
 al_offset 4096
 bm_offset 36864

 Found reiser filesystem

 This would corrupt existing data.
 If you want me to do this, you need to zero out the first part
 of the device (destroy the content).
 You should be very sure that you mean it.
 Operation refused.

 after checking the metadata lv on node3 i saw that it contained a
 filesystem but it was too small for the fs. to find out what the lv
 contains i resized it and mounted it.

 it was an test instance i've deleted before trying to change the disk
 template. i then manually removed the volumes on node 3 and retried the
 disk template change but got the same problem.

 it looks like the new volume was created at the same start as the
 volume from the old instance. 


<b>What is the expected output? What do you see instead?</b>

ganeti should change the disk template from plain to drbd without error or at least should roll back to a usable instance config after this error.

if the process of changing the disk template failes (because of the above error) the instance volumes are named differently and the instnace failes to start again. one needs to rename the volume to the old name (get via gnt-instance info <instance>) before starting the instance. 

<b>Please provide any additional information below.</b>

```
Originally added on 2011-08-01 12:28:08 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
original thread for reference:

http://groups.google.com/group/ganeti/browse_thread/thread/517618626c72b11d/e8f90dff058395c6/drbd
```
Originally added on 2011-08-01 12:29:17 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2011-08-05 13:52:30 +0000 UTC.

Changed State: Started
Added to Milestone: Release2.5

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 18e4dee
Author: Iustin Pop <iustin@google.com>
Date:   Fri Aug 5 15:41:28 2011 +0200

    Zero DRBD metadata before creation

This has been implemented and will be in the 2.5 release. Thanks!
```
Originally added on 2011-08-05 13:59:39 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
do your changes only fix the zeroing issue or also the error handling (wrong volume names after error) if something goes wrong?

```
Originally added on 2011-08-11 07:41:25 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I haven't see a more detailed description of the "wrong volume names after error" (in the email thread it is just "the instance volumes are named differently and the instnace failes to start again", but I think that error is already fixed in 2.4.3, via the commit http://git.ganeti.org/?p=ganeti.git;a=commit;h=db8e5f1ccf21985b97663b8929c56412546d6324

If that still doesn't fix it, please open a new bug for that issue.

thanks,
iustin
```
Originally added on 2011-08-11 10:04:13 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Migrate old issues from 'fixed' to 'released'.
```
Originally added on 2012-10-22 09:16:17 +0000 UTC.

Changed State: Released
